medium-tests/src/test/java/mediumtest/analysis/AnalysisMediumTests.java

```diff
@@ -252,6 +253,56 @@ void it_should_analyze_xml_file_in_connected_mode(SonarLintTestHarness harness,
     assertThat(raisedIssueDto.getRuleKey()).isEqualTo("xml:S3421");
   }
 
+  @SonarLintTest
+  void it_should_analyze_xml_file_after_binding(SonarLintTestHarness harness, @TempDir Path baseDir) {
+    var xmlFileContent = """
+      <?xml version="1.0" encoding="UTF-8"?>
+      <project>
+        <modelVersion>4.0.0</modelVersion>
+        <groupId>com.foo</groupId>
+        <artifactId>bar</artifactId>
+        <version>${pom.version}</version>
+      </project>""";
+    var firstFilePath = createFile(baseDir, "pom.xml", xmlFileContent);
+    var firstFileUri = firstFilePath.toUri();
+    var secondFilePath = createFile(baseDir, "pom.xml", xmlFileContent);
+    var secondFileUri = secondFilePath.toUri();
+    final var OTHER_CONFIG_SCOPE_ID = CONFIG_SCOPE_ID + "2";
+    var client = harness.newFakeClient()
+      .withInitialFs(CONFIG_SCOPE_ID, baseDir, List.of(new ClientFileDto(firstFileUri, baseDir.relativize(firstFilePath), CONFIG_SCOPE_ID, false, null, firstFilePath, null, null, true)))
+      .withInitialFs(OTHER_CONFIG_SCOPE_ID, baseDir, List.of(new ClientFileDto(secondFileUri, baseDir.relativize(secondFilePath), OTHER_CONFIG_SCOPE_ID, false, null, secondFilePath, null, null, true)))
+      .build();
+    var server = harness.newFakeSonarQubeServer()
+      .withProject("projectKey")
+      .start();
+    var backend = harness.newBackend()
+      .withSonarQubeConnection("connectionId", server,
+        storage -> storage.withPlugin(TestPlugin.XML).withProject("projectKey",
+          project -> project.withMainBranch("main").withRuleSet("xml", ruleSet -> ruleSet.withActiveRule("xml:S3421", "BLOCKER"))))
+      .withBoundConfigScope(CONFIG_SCOPE_ID, "connectionId", "projectKey")
+      .withExtraEnabledLanguagesInConnectedMode(Language.XML)
+      .start(client);
+    backend.getFileService().didOpenFile(new DidOpenFileParams(CONFIG_SCOPE_ID, firstFileUri));
+    await().untilAsserted(() -> assertThat(client.getRaisedIssuesForScopeId(CONFIG_SCOPE_ID)).isNotEmpty());
```

Comment:
`client.getRaisedIssuesForScopeId(CONFIG_SCOPE_ID)` returns map URIs to list of issues. So it can contain an entry with a URI key and empty list as value and pass this assertion.
Look like here `client.getRaisedIssuesForScopeIdAsList(CONFIG_SCOPE_ID)` is more appropriate.